tracer/src/Datadog.Trace/DataStreamsMonitoring/DataStreamsManager.cs

```diff
@@ -0,0 +1,134 @@
+﻿// <copyright file="DataStreamsManager.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.DataStreamsMonitoring.Hashes;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Headers;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Vendors.Serilog.Events;
+
+namespace Datadog.Trace.DataStreamsMonitoring;
+
+/// <summary>
+/// Manages all the data streams monitoring behaviour
+/// </summary>
+internal class DataStreamsManager
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<DataStreamsManager>();
+    private readonly NodeHashBase _nodeHashBase;
+    private bool _isEnabled;
+
+    public DataStreamsManager(
+        bool enabled,
+        string env,
+        string defaultServiceName)
+    {
+        _isEnabled = enabled;
+        // We don't yet support primary tag in .NET yet
```

Comment:
Out of curiosity,  what's the impact on the feature?

Reply:
If we had a primary tag, it would be used as part of the hashes, and would be sent in the data payload.

Reply:
very few customers set they primary tag that way, so it shouldn't be an issue.

Reply:
For us, it's useful to detect cross az traffic